bgpd/bgp_attr.c

```diff
@@ -1944,7 +1945,18 @@ bgp_attr_ext_communities(struct bgp_attr_parser_args *args)
 	bgp_attr_evpn_na_flag(attr, &attr->router_flag);
 
 	/* Extract the Rmac, if any */
-	bgp_attr_rmac(attr, &attr->rmac);
+	if (bgp_attr_rmac(attr, &attr->rmac)) {
```

Comment:
It is unnecessary to call these two functions (bgp_attr_rmac() & bgp_mac_exists() )for every packet parsing. In fact, if debug is not enabled, calling these two functions is completely unnecessary. This code can be removed. If you think it is really required to have this debug, at least restructure this code such that these two functions will be called only when debugs are enabled.

Reply:
Changed it.